service/simpleSecretService_test.go

```diff
@@ -0,0 +1,207 @@
+const valFromStorage = "valueFromStorage"
+const password = "passwordIs32bytelongandOtherTxt"
+const read = "Read"
+
+type mockStorage struct{ mock.Mock }
+
+func newMockStorage() *mockStorage { return &mockStorage{} }
+
+func (m *mockStorage) Read(key string) (string, error) {
+	arg := m.Called(key)
+	return arg.String(0), arg.Error(1)
+}
+
+func (m *mockStorage) Write(key string, value string) error {
+	arg := m.Called(key, value)
+	return arg.Error(0)
+}
+
+type mockEncoder struct{ mock.Mock }
+
+func newMockEncoder() *mockEncoder { return &mockEncoder{} }
+
+func (m *mockEncoder) Encrypt(plaintext string, cipherKey []byte) (string, error) {
+	arg := m.Called(plaintext, cipherKey)
+	return arg.String(0), arg.Error(1)
+}
+func (m *mockEncoder) Decrypt(ct string, cipherKey []byte) (string, error) {
+	arg := m.Called(ct, cipherKey)
+	return arg.String(0), arg.Error(1)
+}
+
+func TestSimpleSecretService_WriteSecret2(t *testing.T) {
+	var testCases = []struct {
+		name           string
+		encryptVal     string
+		encryptErr     error
+		value          string
+		writeErr       error
+		secretKey      string
+		key            string
+		numberOfCallsW int
+		numberOfCallsE int
+		expectedErr    bool
+	}{
+		{
+			name:           "Successful Write Secret",
+			encryptVal:     secretVal,
+			secretKey:      password,
+			key:            "key",
+			value:          notSecretVal,
+			numberOfCallsW: 1,
+			numberOfCallsE: 2,
+			expectedErr:    false,
+		},
+		{
+			name:           "Unsuccessful encrypt Write Secret",
+			encryptVal:     secretVal,
+			secretKey:      password,
+			key:            "key",
+			value:          notSecretVal,
+			encryptErr:     errors.New("encrypt Error"),
+			numberOfCallsW: 0,
+			numberOfCallsE: 1,
+			expectedErr:    true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			me := newMockEncoder()
+			defer me.AssertNumberOfCalls(t, "Encrypt", tc.numberOfCallsE)
+			me.On("Encrypt", tc.value, ciphering.PassToSecretKey(tc.secretKey)).Return(tc.encryptVal, tc.encryptErr)
+			me.On("Encrypt", tc.key, ciphering.PassToSecretKey(tc.secretKey)).Return(tc.encryptVal, tc.encryptErr)
+
+			ms := newMockStorage()
+			defer ms.AssertNumberOfCalls(t, "Write", tc.numberOfCallsW)
+			ms.On("Write", tc.encryptVal, tc.encryptVal).Return(tc.writeErr)
+
+			ss := New(ms, me, me)
+			err := ss.WriteSecret("key", tc.value, tc.secretKey)
+			if tc.expectedErr {
+				require.Error(t, err)
+
+			} else {
+				require.Nil(t, err)
+			}
+		})
+	}
+}
+
+func TestSimpleSecretService_ReadSecret(t *testing.T) {
+
+	var testCases = []struct {
+		name           string
+		readVal        string
+		readErr        error
+		encryptVal     string
+		encryptErr     error
+		decryptVal     string
+		decryptErr     error
+		secretKey      string
+		key            string
+		value          string
+		expectedResult string
+		expectedError  bool
+		numberOfCallsR int
+		numberOfCallsE int
+		numberOfCallsD int
```

Comment:
Do you feel comfortable reading these shortcuts? Please rename it to a full words